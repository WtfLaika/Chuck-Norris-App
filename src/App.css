html {
  font-size: 16px !important;
  font-family: 'Raleway', sans-serif;
}

body {
  width: 100%;
}
header {
  background-color: #422ed4;
}

.header-container {
  margin: 0 auto;
  width: 78%;
  display: flex;
  justify-content: left;
  align-items: center;
  color: white;
  font-weight: 800;
  align-self: center;
  height: 100%;
}
.btn-header {
  margin-left: 2%;
  background-color: #422ed4;
  border-width: 0;
  color: white;
  font-size: 16px !important;
  font-family: 'wild-family';
  padding: 16px 10px;
  text-decoration: none;
}
.btn-header:hover {
  background-color: #6655d4;
}
.btn-header:first-of-type {
  margin-left: 10%;
}
.btn-header.btn-header-slct {
  margin-top: 4px;
  border-bottom: 4px solid #150b51;
  height: 100%;
}
.chuck-logo {
  margin-right: 10px;
  height: 43px;
  width: 43px;
}
.categories {
  width: 78%;
  margin: 0 auto;
}

.title {
  margin: 30px 0;
  font-size: 3.125rem;
  font-weight: 800;
  text-align: center;
}
.categories-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-content: space-between;
  z-index: 1;
}
.btn-light {
  height: 57px;

  background-color: white;
  border-radius: 20px;
  color: black;
  width: 15.1%;
  margin-right: 10px;
  font-weight: 800;
  font-size: 1rem !important;
  border: white;
  box-shadow: 0 0 7px 0px rgb(215 206 206);
  margin-bottom: 15px;
}

.btn-light:hover {
  background-color: #422ed4;
  color: white;
}

.phrase-container {
  margin: 0 auto;
  margin-top: -30px;
  width: 78%;
  max-width: 600px;
  display: flex;
  justify-content: flex-end;
  flex-wrap: wrap;
}

#chuck_img {
  position: relative;
  top: 20px;
  z-index: 1;
  margin-right: 20px;
}

.phrase {
  padding: 24px;
  min-height: 100px;
  color: white;
  background-color: #422ed4;
  font-weight: 600;
  font-size: 1.25rem;
  line-height: 1.9rem;
  z-index: 2;
  width: 100%;
  border-radius: 25px;
  position: relative;
  height: max-content;
  top: -85px;
}
.meme {
  width: 100%;
  max-height: 400px;
}
.loaderContainer {
  position: absolute;
  top: 50%;
  left: 50%;
  z-index: 2;
}
