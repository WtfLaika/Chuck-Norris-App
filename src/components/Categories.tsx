import React from 'react';
import {Category} from './Category';

interface ICategoryProps {
  type: ECategoriesTypes;
  categories: string[];
}

export enum ECategoriesTypes {
  JOKES = 'Jokes',
  MEMES = 'Memes',
}

export const Categories = ({type, categories}: ICategoryProps) => {
  return (
    <div className="categories">
      <div className="title">{type} Categories</div>
      <div className="categories-container">
        {categories.map((category, index) => {
          return <Category key={category} category={category} type={type} />;
        })}
      </div>
    </div>
  );
};
