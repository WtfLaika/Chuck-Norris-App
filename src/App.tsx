import React, {useLayoutEffect} from 'react';
import './App.css';
import {Provider} from 'react-redux';
import {createReduxStore} from './redux/rootReducer';
import MainService from './services/main.service';
import {FullScreenLoader} from './components/loader';
import {AppRouter} from './router/appRouter';
import {BrowserRouter} from 'react-router-dom';
import {Header} from './components/Header';
import {setCategories} from './redux/answerSlice';

const {REACT_APP_JOKES_URL, REACT_APP_CATEGORIES} = process.env;

export const store = createReduxStore();
export const App = () => {
  useLayoutEffect(() => {
    MainService.requestGetData(`${REACT_APP_JOKES_URL}${REACT_APP_CATEGORIES}`, setCategories, '');
  }, []);

  return (
    <div className="container">
      <Provider store={store}>
        <BrowserRouter>
          <Header />
          <AppRouter />
          <FullScreenLoader />
        </BrowserRouter>
      </Provider>
    </div>
  );
};

export default App;
