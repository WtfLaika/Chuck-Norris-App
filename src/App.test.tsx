import React from 'react';
import {render} from '@testing-library/react';
import {Provider} from 'react-redux';
import App, {store} from './App';
import {Header} from './components/Header';
import {Category} from './components/Category';
import {Categories} from './components/Categories';
import {ECategoriesTypes} from './components/Categories';
import {renderWithRedux} from './test-helpers/renderWithRedux';
import {Footer} from './components/Footer';
import {renderWithRouter} from './test-helpers/renderWithRouter';
import {rootReducerInitialState} from './redux/rootReducer';

test('renders learn react link', async () => {
  const {getByText} = render(
    <Provider store={store}>
      <App />
    </Provider>,
  );

  expect(getByText(/Chuck Norris/i)).toBeInTheDocument();
});
test('should render Header component', async () => {
  const component = await renderWithRouter(<Header />, {route: '/', initialState: rootReducerInitialState});

  expect(component).toMatchSnapshot();
});

describe('should render Category component', () => {
  it('should render Category with joke ', async () => {
    const category: string = 'animal';
    const component = await renderWithRedux(<Category category={category} type={ECategoriesTypes.JOKES} />);

    expect(component).toMatchSnapshot();
  });
  it('should render Category with joke ', async () => {
    const category: string = 'memes';
    const component = await renderWithRedux(<Category category={category} type={ECategoriesTypes.MEMES} />);

    expect(component).toMatchSnapshot();
  });
});

describe('should render Categories component', () => {
  it('should render Categories with jokes ', async () => {
    const categories = ['animal,career,celebrity'];
    const component = await renderWithRedux(<Categories categories={categories} type={ECategoriesTypes.JOKES} />);

    expect(component).toMatchSnapshot();
  });
  it('should render Categories with memes', async () => {
    const memes = ['memes,dankmemes,random'];
    const component = await renderWithRedux(<Categories categories={memes} type={ECategoriesTypes.MEMES} />);
    expect(component).toMatchSnapshot();
  });
});

describe('should render Footer component', () => {
  it('should render Footer with joke   ', async () => {
    const joke = 'ahhahah something so funny';
    const component = await renderWithRedux(<Footer data={joke} type={ECategoriesTypes.JOKES} />);
    expect(component).toMatchSnapshot();
  });
  it('should render Footer with meme', async () => {
    const memeImg = 'https://static01.nyt.com/images/2021/04/30/multimedia/30xp-meme/29xp-meme-mediumSquareAt3X-v5.jpg';
    const component = await renderWithRedux(<Footer data={memeImg} type={ECategoriesTypes.MEMES} />);
    expect(component).toMatchSnapshot();
  });
});

describe('should render App component', () => {
  it('should render App  without props ', async () => {
    const component = await render(<App />);
    expect(component).toMatchSnapshot();
  });
});
