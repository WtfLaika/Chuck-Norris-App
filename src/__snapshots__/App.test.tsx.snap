// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App component should render App  without props  1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <header>
          <div
            class="header-container"
          >
            <img
              alt="chuck"
              class="chuck-logo"
              src=""
            />
            <div>
              Chuck Norris
            </div>
            <a
              class="btn-header btn-header-slct"
              data-testid="main-link"
              href="/"
            >
              Jokes
            </a>
            <a
              class="btn-header false"
              data-testid="memes-link"
              href="/memes"
            >
              Memes
            </a>
          </div>
        </header>
        <div
          data-testid="main-page"
        >
          <div
            class="categories"
          >
            <div
              class="title"
            >
              Jokes
               Categories
            </div>
            <div
              class="categories-container"
            />
          </div>
          <div
            class="phrase-container"
          >
            <img
              alt="Chuck"
              id="chuck_img"
              src=""
            />
            <div
              class="phrase"
            >
              Choose a catagory to get Chuck's phrase
            </div>
          </div>
        </div>
        <svg
          aria-busy="true"
          aria-label="blocks-loading"
          class="loaderContainer"
          data-testid="color-ring-svg"
          height="80"
          preserveAspectRatio="xMidYMid"
          role="status"
          viewBox="0 0 100 100"
          width="80"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <defs>
            <mask
              id="ldio-4offds5dlws-mask"
            >
              <circle
                cx="50"
                cy="50"
                r="26"
                stroke="#fff"
                stroke-dasharray="122.52211349000194 40.840704496667314"
                stroke-linecap="round"
                stroke-width="9"
                transform="rotate(198.018 50 50)"
              >
                <animatetransform
                  attributeName="transform"
                  dur="1s"
                  keyTimes="0;1"
                  repeatCount="indefinite"
                  type="rotate"
                  values="0 50 50;360 50 50"
                />
              </circle>
            </mask>
          </defs>
          <g
            mask="url(#ldio-4offds5dlws-mask)"
          >
            <rect
              fill="#e15b64"
              height="100"
              width="15"
              x="14.5"
              y="0"
            >
              <animate
                attributeName="fill"
                begin="-0.8s"
                dur="1s"
                keyTimes="0;0.25;0.5;0.75;1"
                repeatCount="indefinite"
                values="#e15b64;#f47e60;#f8b26a;#abbd81;#849b87"
              />
            </rect>
            <rect
              fill="#f47e60"
              height="100"
              width="15"
              x="28.5"
              y="0"
            >
              <animate
                attributeName="fill"
                begin="-0.6s"
                dur="1s"
                keyTimes="0;0.25;0.5;0.75;1"
                repeatCount="indefinite"
                values="#e15b64;#f47e60;#f8b26a;#abbd81;#849b87"
              />
            </rect>
            <rect
              fill="#f8b26a"
              height="100"
              width="15"
              x="42.5"
              y="0"
            >
              <animate
                attributeName="fill"
                begin="-0.4s"
                dur="1s"
                keyTimes="0;0.25;0.5;0.75;1"
                repeatCount="indefinite"
                values="#e15b64;#f47e60;#f8b26a;#abbd81;#849b87"
              />
            </rect>
            <rect
              fill="#abbd81"
              height="100"
              width="15"
              x="56.5"
              y="0"
            >
              <animate
                attributeName="fill"
                begin="-0.2s"
                dur="1s"
                keyTimes="0;0.25;0.5;0.75;1"
                repeatCount="indefinite"
                values="#e15b64;#f47e60;#f8b26a;#abbd81;#849b87"
              />
            </rect>
            <rect
              fill="#849b87"
              height="100"
              width="15"
              x="70.5"
              y="0"
            >
              <animate
                attributeName="fill"
                begin="0s"
                dur="1s"
                keyTimes="0;0.25;0.5;0.75;1"
                repeatCount="indefinite"
                values="#e15b64;#f47e60;#f8b26a;#abbd81;#849b87"
              />
            </rect>
          </g>
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <header>
        <div
          class="header-container"
        >
          <img
            alt="chuck"
            class="chuck-logo"
            src=""
          />
          <div>
            Chuck Norris
          </div>
          <a
            class="btn-header btn-header-slct"
            data-testid="main-link"
            href="/"
          >
            Jokes
          </a>
          <a
            class="btn-header false"
            data-testid="memes-link"
            href="/memes"
          >
            Memes
          </a>
        </div>
      </header>
      <div
        data-testid="main-page"
      >
        <div
          class="categories"
        >
          <div
            class="title"
          >
            Jokes
             Categories
          </div>
          <div
            class="categories-container"
          />
        </div>
        <div
          class="phrase-container"
        >
          <img
            alt="Chuck"
            id="chuck_img"
            src=""
          />
          <div
            class="phrase"
          >
            Choose a catagory to get Chuck's phrase
          </div>
        </div>
      </div>
      <svg
        aria-busy="true"
        aria-label="blocks-loading"
        class="loaderContainer"
        data-testid="color-ring-svg"
        height="80"
        preserveAspectRatio="xMidYMid"
        role="status"
        viewBox="0 0 100 100"
        width="80"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <defs>
          <mask
            id="ldio-4offds5dlws-mask"
          >
            <circle
              cx="50"
              cy="50"
              r="26"
              stroke="#fff"
              stroke-dasharray="122.52211349000194 40.840704496667314"
              stroke-linecap="round"
              stroke-width="9"
              transform="rotate(198.018 50 50)"
            >
              <animatetransform
                attributeName="transform"
                dur="1s"
                keyTimes="0;1"
                repeatCount="indefinite"
                type="rotate"
                values="0 50 50;360 50 50"
              />
            </circle>
          </mask>
        </defs>
        <g
          mask="url(#ldio-4offds5dlws-mask)"
        >
          <rect
            fill="#e15b64"
            height="100"
            width="15"
            x="14.5"
            y="0"
          >
            <animate
              attributeName="fill"
              begin="-0.8s"
              dur="1s"
              keyTimes="0;0.25;0.5;0.75;1"
              repeatCount="indefinite"
              values="#e15b64;#f47e60;#f8b26a;#abbd81;#849b87"
            />
          </rect>
          <rect
            fill="#f47e60"
            height="100"
            width="15"
            x="28.5"
            y="0"
          >
            <animate
              attributeName="fill"
              begin="-0.6s"
              dur="1s"
              keyTimes="0;0.25;0.5;0.75;1"
              repeatCount="indefinite"
              values="#e15b64;#f47e60;#f8b26a;#abbd81;#849b87"
            />
          </rect>
          <rect
            fill="#f8b26a"
            height="100"
            width="15"
            x="42.5"
            y="0"
          >
            <animate
              attributeName="fill"
              begin="-0.4s"
              dur="1s"
              keyTimes="0;0.25;0.5;0.75;1"
              repeatCount="indefinite"
              values="#e15b64;#f47e60;#f8b26a;#abbd81;#849b87"
            />
          </rect>
          <rect
            fill="#abbd81"
            height="100"
            width="15"
            x="56.5"
            y="0"
          >
            <animate
              attributeName="fill"
              begin="-0.2s"
              dur="1s"
              keyTimes="0;0.25;0.5;0.75;1"
              repeatCount="indefinite"
              values="#e15b64;#f47e60;#f8b26a;#abbd81;#849b87"
            />
          </rect>
          <rect
            fill="#849b87"
            height="100"
            width="15"
            x="70.5"
            y="0"
          >
            <animate
              attributeName="fill"
              begin="0s"
              dur="1s"
              keyTimes="0;0.25;0.5;0.75;1"
              repeatCount="indefinite"
              values="#e15b64;#f47e60;#f8b26a;#abbd81;#849b87"
            />
          </rect>
        </g>
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render Categories component should render Categories with jokes  1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="categories"
      >
        <div
          class="title"
        >
          Jokes
           Categories
        </div>
        <div
          class="categories-container"
        >
          <button
            class="btn-light"
          >
            animal,career,celebrity
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="categories"
    >
      <div
        class="title"
      >
        Jokes
         Categories
      </div>
      <div
        class="categories-container"
      >
        <button
          class="btn-light"
        >
          animal,career,celebrity
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render Categories component should render Categories with memes 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="categories"
      >
        <div
          class="title"
        >
          Memes
           Categories
        </div>
        <div
          class="categories-container"
        >
          <button
            class="btn-light"
          >
            memes,dankmemes,random
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="categories"
    >
      <div
        class="title"
      >
        Memes
         Categories
      </div>
      <div
        class="categories-container"
      >
        <button
          class="btn-light"
        >
          memes,dankmemes,random
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render Category component should render Category with joke  1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="btn-light"
      >
        animal
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="btn-light"
    >
      animal
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render Category component should render Category with joke  2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="btn-light"
      >
        memes
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="btn-light"
    >
      memes
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render Footer component should render Footer with joke    1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="phrase-container"
      >
        <img
          alt="Chuck"
          id="chuck_img"
          src=""
        />
        <div
          class="phrase"
        >
          ahhahah something so funny
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="phrase-container"
    >
      <img
        alt="Chuck"
        id="chuck_img"
        src=""
      />
      <div
        class="phrase"
      >
        ahhahah something so funny
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render Footer component should render Footer with meme 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="phrase-container"
      >
        <img
          alt="Chuck"
          id="chuck_img"
          src=""
        />
        <div
          class="phrase"
        >
          <img
            class="meme"
            src="https://static01.nyt.com/images/2021/04/30/multimedia/30xp-meme/29xp-meme-mediumSquareAt3X-v5.jpg"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="phrase-container"
    >
      <img
        alt="Chuck"
        id="chuck_img"
        src=""
      />
      <div
        class="phrase"
      >
        <img
          class="meme"
          src="https://static01.nyt.com/images/2021/04/30/multimedia/30xp-meme/29xp-meme-mediumSquareAt3X-v5.jpg"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render Header component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="main-page"
      >
        <div
          class="categories"
        >
          <div
            class="title"
          >
            Jokes
             Categories
          </div>
          <div
            class="categories-container"
          />
        </div>
        <div
          class="phrase-container"
        >
          <img
            alt="Chuck"
            id="chuck_img"
            src=""
          />
          <div
            class="phrase"
          >
            Choose a catagory to get Chuck's phrase
          </div>
        </div>
      </div>
      <header>
        <div
          class="header-container"
        >
          <img
            alt="chuck"
            class="chuck-logo"
            src=""
          />
          <div>
            Chuck Norris
          </div>
          <a
            class="btn-header btn-header-slct"
            data-testid="main-link"
            href="/"
          >
            Jokes
          </a>
          <a
            class="btn-header false"
            data-testid="memes-link"
            href="/memes"
          >
            Memes
          </a>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="main-page"
    >
      <div
        class="categories"
      >
        <div
          class="title"
        >
          Jokes
           Categories
        </div>
        <div
          class="categories-container"
        />
      </div>
      <div
        class="phrase-container"
      >
        <img
          alt="Chuck"
          id="chuck_img"
          src=""
        />
        <div
          class="phrase"
        >
          Choose a catagory to get Chuck's phrase
        </div>
      </div>
    </div>
    <header>
      <div
        class="header-container"
      >
        <img
          alt="chuck"
          class="chuck-logo"
          src=""
        />
        <div>
          Chuck Norris
        </div>
        <a
          class="btn-header btn-header-slct"
          data-testid="main-link"
          href="/"
        >
          Jokes
        </a>
        <a
          class="btn-header false"
          data-testid="memes-link"
          href="/memes"
        >
          Memes
        </a>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
